### DISCUSSION TITLE
Using unsecure ImageRepository

### DISCUSSION THREAD

**Discussion by mvanderkroon** [fluxcd/flux2#3900] [General] [answered]
Currently ImageRepository only accepts secured image repositories by automatically prefixing the `https` section of the image repository. This, of course, makes a lot of sense for any real world / production scenario, however as these things go I have found a bonafide use for 'unsecured' image repositories: local development.

As it is, I'm trying to set up a k3d cluster with a local registry where Flux manages the deployments. The idea is to be able to locally build and push containers (to the local registry) and have Flux automatically pick these up using hte Image Automation. Unfortunately this doesn't work, as the local registry is unsecure and thus cannot be read from by the controller.

This approach seems to gel at least somewhat with kubernetes, as kubernetes does allow unsecured registries in the `image` field.

Ultimately, I'd like to be able to develop different components of my platform/application locally, integrating the lot using GitOps/Flux locally too before pushing to a *secure* image repository for the production cluster (also managed by Flux) to pick up.

Either way, I'm curious to see if anyone has found a way to  accomplish the above, or perhaps even to consider adding the ability to use unsecured image repositories to the image automation controller.

**Comment by kingdonb** [comment, marked as answer]
There is an RFC for insecure registries that's now work-in-progress here, please follow that issue for comments and progress/status updates:

https://github.com/fluxcd/image-reflector-controller/pull/395
